csi/util.go

```diff
@@ -239,6 +239,26 @@ func parseJSONRecurringJobs(jsonRecurringJobs string) ([]longhornclient.Recurrin
 	return recurringJobs, nil
 }
 
+func syncMountPointDirectory(targetPath string) error {
+	d, err := os.Open(targetPath)
+	if err != nil {
+		return err
+	}
+	defer d.Close()
+
+	if _, err := d.Readdirnames(-1); err != nil {
+		return err
+	}
+
+	// it would not always return `Input/Output Error` or `read-only file system` errors if we only use Sync() without the previous I/O operation
+	// and it needs more investigations why only Sync() would not return an error
+	if err := d.Sync(); err != nil {
+		return err
+	}
```

Comment:
Have the same concern as well. Modified to create an empty file and remove it to check if the mount point is valid.
Thanks

Reply:
The `return f.Sync()` function below is still sync the host filesystem, right? Can we remove the sync and still be able to detect the IO error?

Reply:
No, it only synchronize the temp file just created.
I don't have any other idea that how could it detect the IO error.
Any good suggestion?

Reply:
What if there are lots of caches that haven't been flushed before the volume crash? Will this late sync lead to any unexpected issues?

As I mentioned before, can we try to use the direct IO libs in `sparse-tools` to modify the disk meta-file instead?

Reply:
Do you mean access the volume directly instead of accessing the file system?

Reply:
If the filesystem is somehow in `readonly` state, will the error lead to any side effect?

Reply:
It might get into the `NodeStageVolume/NodeUnStageVolume` loop

Reply:
I see. Should we ignore the error and make it as best-effort instead?

Reply:
ignore `read-only file system`? let me do some testings for ignoring this error.

Reply:
@mantissahz 
After going through the usage of `ensureMountPoint()`, I think returning error is fine and no need to ignore it.

Reply:
@derekbit files in the directory of the mount point `globalmount` with the `read-only file system` error for opening a file would still return `Input/output error` when accessing.

Reply:
Do you mean that there is no need to write anything or run `sync` to trigger the IO error if the fs is corrupted?

Reply:
No. It still needs some I/Os to trigger I/O error. What I meant is files might not be accessible if they are in a read-only file system for the situation of this issue.

Reply:
Then what I mentioned above may be applicable:
You can directly open/create the disk meta file `<disk mount point or data path>/longhorn-disk.cfg` directly launch a new file like `<disk mount point or data path>/longhorn-disk.tmp`.

Reply:
Yes, now I create a temp empty file in the mount point directory to test the file system and remove it after testing without `sparse-tools` libs but `OpenFile` with the `O_SYNC` flag instead of synchronizing the whole file system.
@shuo-wu Does it look like what you mentioned?

Reply:
I think it's good enough to use `O_SYNC`.